example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java

```diff
@@ -7,59 +7,104 @@
 import static com.jayway.restassured.RestAssured.given;
 import static com.jayway.restassured.RestAssured.when;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 
-import java.util.ArrayList;
-import java.util.List;
+/**
+ * Created by Tamas_Csako
+ */
+public class BasicServiceTest {
+    private static Logger log = Logger.getLogger(BasicServiceTest.class.getName());
+    private static final int NUMBER_OF_RESPONSE = 1;
+    private static final String CONTENT_NUMBER_OF_ELEMENTS = "numberOfElements";
+    private static final String CONTENT_EMAIL_ADDRESS = "content.emailAddress";
+    private static final int HTTP_OK = 200;
 
-import org.junit.Before;
-import org.junit.Test;
+    // Test data
+    private String firstName;
+    private String lastName;
+    private String emailAddress;
+    private String emailAddressConfirmation;
+    private boolean subscribeNewsletter;
 
-public class BasicServiceTest {
-	//Test data
-	private String firstName;
-	private String lastName;
-	private String emailAddress;
-	private String emailAddressConfirmation;
-	private String newsletterOptIn;
+    // Verification
+    private List<String> listToVerifyEmail;
+    private HomePageObject homePageObject;
+    private ThankYouPageObject thankYouPageObject;
+    private ThankYouPageVerifier thankYouPageVerifier;
+    private WebDriver driver;
+
+    /**
+     * Responsible for setting up test data and environment.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
+
+        log.info("*******************************************");
+        log.info("Deleting existing records");
+        given().delete(ServiceTestingProperties.REST_API_URL);
 
-	//Verification
-	private List<String> listToVerifyEmail;
-	private static final int NUMBER_OF_RESPONSE = 1;
-	private static final String CONTENT_NUMBER_OF_ELEMENTS = "numberOfElements";
-	private static final String CONTENT_EMAIL_ADDRESS = "content.emailAddress";
-	private static final int HTTP_OK = 200;
+        log.info("Setting up test data");
+        firstName = "John";
+        lastName = "Doe";
+        emailAddress = "johndoe@freecloud.com";
+        emailAddressConfirmation = "johndoe@freecloud.com";
+        subscribeNewsletter = true;
 
-	@Before
-	public void setUp() throws Exception {
-		//Delete all existing record
-		given().delete(ServiceTestingProperties.REST_API_URL);
-		
-		//Setup test data
-		firstName = "John";
-		lastName = "Doe";
-		emailAddress = "johndoe@freecloud.com";
-		emailAddressConfirmation = "johndoe@freecloud.com";
-		newsletterOptIn = "true";
-		
-		//Setup verification data
-		listToVerifyEmail = new ArrayList<String>();
-		listToVerifyEmail.add(emailAddress);
-	}
+        log.info("Setting up verification data");
+        listToVerifyEmail = new ArrayList<String>();
+        listToVerifyEmail.add(emailAddress);
+        log.info("Initializing Firefox driver");
+        driver = new FirefoxDriver();
```

Comment:
create abstract webdriver test class and define the before method there to initialize web driver object